src/exceptions.py

```diff
@@ -66,3 +66,35 @@ def __init__(self, msg: str):
             msg (str): Explanation of the error.
         """
         self.msg = msg
+
+
+class PeerRelationUnavailableError(Exception):
```

Comment:
Considering these two exceptions belong to the charm domain, I'd rather define them in the `charm.py` file

Reply:
I'm still thinking that they have meaning only by their name and so I would like to keep their code separated as it doesn't help in anything to understand what the charm does.

Reply:
What do you mean? Those are related to charm conditions. If we separate them, it makes the code less cohesive

Reply:
When I'm reading the charm code, those exceptions' code don't add any value in understanding it. They are interesting only by their name (and that's the only thing that differentiate them). In that sense, it makes sense to me to regroup all those names (a bit like constants, they merely exist, they have no logic) in a file

Reply:
I get your point, but all the charm code will end up depending on the `exceptions`module, which is not ideal IMHO

Reply:
Anyway, this is a minor detail

Reply:
I moved all the exceptions to their corresponding module. The only thing I don't like is that I can't automatically prefix an exceptions with `exceptions.`, I have to think where it comes from.
But I do agree that it makes sense logically that they are with their module. Let's try it like this !